src/coreclr/jit/block.h

```diff
@@ -666,6 +669,38 @@ struct BasicBlock : private LIR::Range
         SetJumpKindAndTarget(jumpKind, jumpDest DEBUG_ARG(compiler));
     }
 
+    BasicBlock* GetFallThroughSucc() const
+    {
+        assert(bbFallsThrough());
+        // For now, bbFallThroughSucc cannot diverge from bbNext
+        assert(FallsIntoNext());
+        return bbFallThroughSucc;
+    }
+
+    void SetFallThroughSucc(BasicBlock* fallThroughSucc)
+    {
+        // TODO: Once we allow bbFallThroughSucc to diverge from bbNext, ensure we only set bbFallThroughSucc
+        // to something other than bbNext when this block is a BBJ_COND.
+        // BBJ_CALLFINALLY can fall through too, but we don't allow its fallthrough successor to diverge from bbNext
+        // because we aren't interested in splitting up call-always pairs
```

Comment:
This is a confusing comment. `BBJ_CALLFINALLY` never fall through, as far the flowgraph is concerned - they always "jump" to their corresponding finally handler. The handler then "jumps back" to the `BBJ_ALWAYS` part of the pair via `BBJ_EHFINALLYRET`.

(Physically this can be implemented as a call, so it looks a bit like `BBJ_CALLFINALLY` falls through to `BBJ_ALWAYS`, but that's a very low-level view)

Reply:
I used the "fall through" language there based on the fact that `BasicBlock::bbFallsThrough` returns true for `BBJ_CALLFINALLY` blocks part of `BBJ_CALLFINALLY/BBJ_ALWAYS` pairs, though I see how describing these blocks as falling through isn't wholly accurate. I can clarify that comment; something like "For `BBJ_CALLFINALLY` blocks part of call-always pairs, `bbFallThroughSucc` points to the `BBJ_ALWAYS` block. For now, call-always pairs are always contiguous, so `bbFallThroughSucc` cannot diverge from `bbNext`."